sources/packages/backend/libs/services/src/sfas/sfas-individual.service.ts

```diff
@@ -63,4 +66,43 @@ export class SFASIndividualService extends DataModelService<SFASIndividual> {
       },
     });
   }
+
+  /**
+   * Gets the total number of unsuccessful completion weeks for a student in SFAS.
+   * @param studentId student id to retrieve the unsuccessful weeks.
+   * @returns total number of unsuccessful completion weeks for a student in SFAS.
+   */
+  async getSFASTotalUnsuccessfulCompletionWeeks(
+    studentId: number,
+  ): Promise<number> {
+    const studentData = await this.studentRepo.findOne({
+      select: {
+        id: true,
+        birthDate: true,
+        sinValidation: { sin: true },
+        user: { lastName: true },
+      },
+      relations: {
+        sinValidation: true,
+        user: true,
+      },
+      where: {
+        id: studentId,
+      },
+    });
+    const sin = studentData.sinValidation.sin;
+    const birthDate = studentData.birthDate;
+    const lastName = studentData.user.lastName;
+    const sfasIndividualData = await this.repo
+      .createQueryBuilder("sfasIndividual")
+      .select(
+        "SUM(sfasIndividual.unsuccessfulCompletion)",
+        "totalUnsuccessfulWeeks",
+      )
+      .where("sin = :sin", { sin })
+      .andWhere("birth_date = :birthDate", { birthDate })
+      .andWhere("LOWER(last_name) = LOWER(:lastName)", { lastName })
+      .getRawOne<SFASIndividualDataSummary>();
+    return sfasIndividualData.totalUnsuccessfulWeeks;
+  }
```

Comment:
`sfasIndividualData?.totalUnsuccessfulWeeks ?? 0`

Reply:
When the student is not available I would recommend returning null here and changing the return type also to reflect it. The null check and possible fallback to zero can happen in the consumer, IMO.